### DISCUSSION TITLE
Data encryption

### DISCUSSION THREAD

**Discussion by splix** [siderolabs/talos#7990] [Q&A] [answered]
I've noticed a couple of questions/issues regarding disk encryption and am wondering about the current thoughts on it.

As I understand it, Telos currently only supports disk encryption for "state" (general Telos configurations) and "ephemeral" data (like emptyDir in pods?). There were some proposals for a "data" type of encryption in issue #4041, and for a general machine disk encryption option in issue #5735. As far as I understand, these were only mentioned in the tickets and it's not implemented yet. Is that correct? Is there any roadmap for this?

Thank you.

**Comment by smira** [comment, marked as answer]
Yes, this is correct. There's a plan to overhaul the storage story with Talos 1.7 early next year

**Reply by splix** [reply]
Thank you. Good to know. Do you think it will be possible to migrate from an unencrypted storage to encrypted one?

**Reply by smira** [reply]
That is always very tough to implement given that Talos reconciles the state to match the spec (configuration), while live encryption is something which should run in the background
